src/Dashboard/Prompty/components/PromptyInitialize.js

```diff
@@ -0,0 +1,93 @@
+import { MySwal } from "../../../Generic/Notify";
+import { getFirestore, doc, getDoc, setDoc } from "firebase/firestore";
+import { useNavigate } from "react-router-dom";
+import { useContext } from "react";
+import { UserContext } from "../../../Auth";
+import { signIn } from "../../../Auth";
+
+async function checkInstance(joinCode) {
+  const db = getFirestore();
+  const docRef = doc(db, "prompty", joinCode);
+  const docSnap = await getDoc(docRef);
+  let data = docSnap.data();
+  console.log(data);
+  if (data !== undefined) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+const PromptyInitialize = () => {
+  const navigate = useNavigate();
+  const { user } = useContext(UserContext);
+
+  const validateAndJoinPrompty = () => {
+    if (user) {
+      console.log(user);
+      joinPrompty(user);
+      return;
+    }
+    // otherwise, open a modal to sign in
+    MySwal.fire({
+      title: "Please sign in to continue",
+      backdrop: true,
+      icon: "info",
+      showCancelButton: true,
+      confirmButtonText: `Sign in with Google`,
+      confirmButtonColor: "#4285f4",
+      cancelButtonText: "Cancel",
+    }).then((result) => {
+      if (!result.isConfirmed) return;
+      signIn().then(() => {});
+    });
+  };
+  function createUserPromptyInstance(joinCode, uid, displayName) {
+    const db = getFirestore();
+    let docRef = doc(db, "prompty", joinCode, "instances", uid);
+    setDoc(docRef, { displayName: displayName }, { merge: true });
+  }
+  const joinPrompty = (userDetails) => {
+    MySwal.fire({
+      title: "Start Prompty",
+      input: "text",
+      inputLabel: "Enter the Prompty code",
+      inputPlaceholder: "071295",
+      inputAttributes: {
+        autocapitalize: "on",
+      },
+      showCancelButton: true,
+      confirmButtonText: "Join",
+      showLoaderOnConfirm: true,
+      inputValidator: async (value) => {
+        let validator = await checkInstance(value);
+        if (validator === true) {
+          createUserPromptyInstance(
+            value,
+            userDetails.uid,
+            userDetails.displayName
+          );
+          navigate("/dash/prompty/" + value);
+        } else {
+          return "Code Invalid!";
+        }
+      },
+    });
+  };
+
+  return (
```

Comment:
can you match the styles to the main datamax page?

<img width="757" alt="datamax page" src="https://github.com/parthsarin/stanford-craft/assets/5355724/5b4e5f72-fdff-4dc6-8c3d-9ad83a83fef6">

(i.e., same color and styles, add a font awesome icon before the start button if possible)